react-components/src/hooks/useClickedNode.tsx

```diff
@@ -44,44 +48,63 @@ export const useClickedNodeData = (): ClickedNodeData | undefined => {
     };
   }, [viewer]);
 
-  const nodeData = useFdm3dNodeData(
+  const nodeDataPromises = useFdm3dNodeDataPromises(
     cadIntersection?.model.modelId,
     cadIntersection?.model.revisionId,
     cadIntersection?.treeIndex
   ).data;
 
   useEffect(() => {
-    if (isWaitingForQueryResult()) {
-      return;
-    }
+    void setClickedNodeFromQueryResult(nodeDataPromises);
 
-    const nodeDataList = nodeData ?? [];
+    async function setClickedNodeFromQueryResult(
+      promises: FdmNodeDataPromises | undefined
+    ): Promise<void> {
+      if (promises === undefined) {
+        return;
+      }
 
-    if (cadIntersection === undefined) {
-      setClickedNodeData(undefined);
-      return;
-    }
+      if (cadIntersection === undefined) {
+        setClickedNodeData(undefined);
+        return;
+      }
+
+      const cadAndFdmNodes = await promises.cadAndFdmNodesPromise;
+
+      if (cadAndFdmNodes === undefined || cadAndFdmNodes.fdmIds.length === 0) {
+        setClickedNodeData({
+          intersection: cadIntersection
+        });
+        return;
+      }
 
-    if (nodeDataList.length === 0) {
       setClickedNodeData({
-        intersection: cadIntersection
+        intersection: cadIntersection,
+        fdmNode: cadAndFdmNodes.fdmIds[0],
+        cadNode: cadAndFdmNodes.cadNode
       });
-      return;
+
+      await setClickedNodeFromViewPromise(cadAndFdmNodes, promises.viewsPromise);
```

Comment:
Is there any reason why we want to split the return in 3 parts and not 1 or 2? Can't we just await for all promises to finish and then set `clickedNodeData` so that the user sees only one data update.

Reply:
Cleared up in DMs